iree/compiler/Utils/CustomKernelsTargetInfo.h

```diff
@@ -33,6 +33,7 @@ enum class CustomKernelTargetFeature {
   Intrinsics,
   // Aarch64 features.
   Aarch64Dotprod,
+  Aarch64I8mm,
 };
 
 inline bool isFeatureForArch(CustomKernelTargetFeature feature,
```

Comment:
Besides Codegen, this is also used by `convert-linalg-matmul-to-mmt4d`, which currently is in Flow. It's incidental that it is in Flow as it's not even used/exposed at all by iree-translate at the moment -- at the moment it's manually used with iree-opt, so we could have it anywhere. What's not incidental though is that it's happening much earlier than Codegen.

Reply:
Is it "earlier than codegen" or just a different form of code generation when compared with the other MLIR/Linalg-based codegen in IREE? It sure looks and sounds like device code generation to me :P

Reply:
It really is "earlier than codegen". too big topic for this comment thread, but we can have a video call with @benvanik and @MaheshRavishankar if we want to get to the bottom of this now (it's been a topic for a while).

Reply:
My vocabulary is pretty limited here - I mostly view anything happening device-side (cpu codegen, register allocation, linalg, fusion) as a black box. The main thing I'm getting at is: this code feels sufficiently special purpose that it deserves a place in the tree more deliberate than a "Utils/" folder. Maybe it's `compiler/Codegen/`, maybe it's `compiler/CustomKernels`, maybe it's [iree/builtins](https://github.com/google/iree/tree/main/iree/builtins), or somewhere else. I'm not sure beyond "not utils".

Reply:
I see. Let's resolve this soon then, maybe @benvanik has an opinion. Since the files there are from a previous PR, I agree this is fair to leave to a future PR.

Reply:
yeah Utils/ is the wrong place for it for sure - this belongs in some CPU-specific Codegen/ path - and that's because the mmt4d stuff that uses this _also_ belongs in Codegen/. Was chatting with gcmn the other day and we're really going to have to start moving on actually breaking this stuff up soon - as I'm seeing more of this at this point I'm becoming increasingly skeptical of the need to continue to do anything but start to land things in the right place.

Since we're just running this pass from iree-opt moving it to codegen/llvmcpu/ seems quick and easy and avoids the implicit (and obfuscated) flow<->codegen cross deps that would be required (putting codegen stuff in Utils/ to share with both codegen and a part of codegen that's in the wrong place is effectively adding a dep between flow and codegen that's just more annoying to sever).

Reply:
We're not just running it from iree-opt anymore. Per our discussion the other day, I added a developer flag to iree-translate to allow flipping this on in flow (https://github.com/google/iree/pull/8402). I agree that we should start work on untangling this though.

Reply:
We shouldn't need a typed enum on the frontend though - strings should be fine there (just like llvm triples and features are passed as strings to avoid needing to cut across all layers like this).